wallet-new/src/Cardano/Wallet/Kernel/Migration.hs

```diff
@@ -2,283 +2,84 @@ module Cardano.Wallet.Kernel.Migration (migrateLegacyDataLayer) where
+-- we leave the node running, we would give the user a chance
 -- to submit a bug report from the Daedalus interface.
 --
 -- However when @forced@ is True the migration is a all-or-nothing.
--- If anything fails (e.g. wallet decoding, resotartion etc) the node crashes.
+-- If anything fails the node crashes.
 migrateLegacyDataLayer :: Kernel.PassiveWallet
-                       -> FilePath
                        -> Bool
                        -> IO ()
-migrateLegacyDataLayer pw unresolvedDbPath forced = do
-    let logMsg = pw ^. Kernel.walletLogMessage
-    logMsg Info "Starting acid state migration"
-    resolved <- resolveDbPath unresolvedDbPath
-    case resolved of
-       Nothing -> -- We assume no migration is needed and we move along
-            case forced of
-                True -> do
-                    logMsg Error $ "Migration failed! Legacy DB was not found at " <> pack unresolvedDbPath <> ", but migration is forced"
-                    exitFailure
-                False -> do
-                    logMsg Info $ "No legacy DB at " <> pack unresolvedDbPath <> ", migration is not needed."
-       Just legacyDbPath -> do
-           bracketLegacyDB legacyDbPath $ \st -> do
-                let  (unavailable, available) = partitionEithers (metadataFromWalletStorage st)
-                     unavailableLen = length unavailable
-                     availableLen   = length available
-
-                when (unavailableLen > 0) $ do
-                    let msg = show unavailableLen
-                              <> " out of "
-                              <> show (unavailableLen + availableLen)
-                              <> " wallets failed to be converted into the metadata "
-                              <> "information needed for migration."
-                    case forced of
-                        True -> do
-                            logMsg Error $ "Migration failed! " <> msg
-                            exitFailure
-                        False -> do
-                            logMsg Error msg
-
-                when (availableLen > 0) $ do
-                    logMsg Info $ "Found "
-                        <> show availableLen
-                        <> " rootAddress(es) to migrate."
-
-                mapM_ (restore pw forced) available
-
-           -- Now that we have closed the DB, we can move the directory
-           backupPath <- moveLegacyDB legacyDbPath
-
-           -- asynchronous restoration still runs at this point.
-           logMsg Info $ "acid state migration succeeded. Old db backup can be found at " <> pack backupPath
+migrateLegacyDataLayer pw forced = do
+    logMsg Info "Starting wallet(s) migration"
+    wKeys <- Keystore.getKeys (pw ^. Kernel.walletKeystore)
+    mapM_ (restore pw forced) wKeys
 
+    -- asynchronous restoration still runs at this point.
+    logMsg Info $ "Migration succeeded, restoration of migrated wallets in progress..."
```

Comment:
We might still want to preserve the backup aspect of the migration though. Especially because, we identify that a migration should be done by looking at the db path, if there's none, we assume it's migrated (or new). 

Therefore, this `moveLegacyDB` is rather important to us :sweat_smile:

Reply:
ah I see! I will bring it back...